gui/src/components/firmware-update/FirmwareUpdate.tsx

```diff
@@ -280,25 +281,29 @@ export function FirmwareUpdate() {
         {}
       ),
     });
-    queueFlashing(devices);
+    setStatus({});
+    clear();
   };
 
   const startUpdate = () => {
     const selectedDevices = Object.keys(selectedDevicesForm)
       .filter((d) => selectedDevicesForm[d])
-      .map((id) => {
+      .reduce((curr, id) => {
         const device = devices.find(({ id: dId }) => id === dId?.id.toString());
 
-        if (!device) throw new Error('no device found');
-        return {
-          type: FirmwareUpdateMethod.OTAFirmwareUpdate,
-          deviceId: id,
-          deviceNames: deviceNames(device, l10n),
-        };
-      });
+        if (!device) return curr;
+        return [
+          ...curr,
+          {
+            type: FirmwareUpdateMethod.OTAFirmwareUpdate,
+            deviceId: id,
+            deviceNames: deviceNames(device, l10n),
+          },
+        ];
+      }, [] as SelectedDevice[]);
```

Comment:
the `if(!device)` check should be on filter function and this should be a `map`

Reply:
edited the review

Reply:
Why reduce is the perfect case to perform both filter and map at the same time. it is literally made for it

Reply:
wait youre right tho ... this is cleaner

Reply:
i need device both for the filter and the content of the map. so no reduce is the best case